src/App.tsx

```diff
@@ -14,6 +19,10 @@ function App() {
     appSessionState,
   });
 
+  useEffect(() => {
+    onStatusUpdate?.(appSessionState.status);
+  }, [appSessionState.status, onStatusUpdate]);
+
```

Comment:
I did not add tests for this functionality. 

I cannot test part of it, for example, the code in `index.tsx` and the code in `App.tsx` is trivial, but testing is not easy because I need to inject session changes in `AppSessionStateContext`.